src/web3client/liquidator.js

```diff
@@ -39,18 +41,26 @@ class Liquidator {
     };
   }
 
-  async isPossibleToClose (superToken, sender, receiver) {
-    // Note: If flow does not exist on the network, we are going to remove from DB
-    return (await this.app.protocol.checkFlow(superToken, sender, receiver)) !== undefined &&
-      (await this.app.protocol.isAccountCriticalNow(superToken, sender));
+  async isPossibleToClose (superToken, sender, receiver, pmode) {
+    const checkFlow = await this.app.protocol.checkFlow(superToken, sender, receiver);
+    const isCritical = await this.app.protocol.isAccountCriticalNow(superToken, sender);
+    if(pmode === 0) {
+      return checkFlow !== undefined && isCritical;
+    } else if(pmode === 1) {
+      const isPatrician = await this.app.protocol.isPatricianPeriodNow(superToken, sender);
+      return checkFlow !== undefined && isCritical && !isPatrician.isPatricianPeriod;
+    } else {
+      const isSolvent = await this.app.protocol.isAccountSolventNow(superToken, sender);
+      return checkFlow !== undefined && isCritical && !isSolvent;
+    }
```

Comment:
Using something resembling enums for p(pp)mode would make it a bit more readable

Reply:
yes, js don't have enum, going to adapt